File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/manager/controller/ViewController.java
package com.trtc.uikit.livekit.manager.controller;

import com.trtc.uikit.livekit.R;
import com.trtc.uikit.livekit.common.utils.ToastUtils;
import com.trtc.uikit.livekit.service.ILiveService;
import com.trtc.uikit.livekit.state.LiveDefine;
import com.trtc.uikit.livekit.state.LiveState;

public class ViewController extends Controller {

    public ViewController(LiveState state, ILiveService service) {
        super(state, service);
    }

    @Override
    public void destroy() {

    }

    public void enableAutoOpenCameraOnSeated(boolean enable) {
        mViewState.autoOpenCameraOnSeated.set(enable);
    }

    public void updateLiveStatus(LiveDefine.LiveStatus status) {
        mViewState.liveStatus.set(status);
    }

    public void onLiveEnd(String roomId) {
        ToastUtils.toast(R.string.livekit_room_destroy);
        updateLiveStatus(LiveDefine.LiveStatus.DASHBOARD);
    }
}

File: Flutter/livekit/lib/manager/controller/view_controller.dart
import 'package:flutter/material.dart';
import 'package:tencent_live_uikit/common/index.dart';
import 'package:tencent_live_uikit/manager/index.dart';
import 'package:tencent_live_uikit/state/index.dart';

class ViewController extends Controller {
  ViewController(super.state, super.service);

  @override
  void destroy() {}

  void finish() {
    Navigator.pop(Global.appContext());
  }

  void enableAutoOpenCameraOnSeated(bool enable) {
    viewState.autoOpenCameraOnSeated.value = enable;
  }

  void updateLiveStatus(LiveStatus status) {
    viewState.liveStatus.value = status;
  }

  void onLiveEnd(String roomId) {
    makeToast(msg: LiveKitLocalizations.of(Global.appContext())!.livekit_room_destroy);
    viewState.liveStatus.value = LiveStatus.dashboard;
  }
}
